### PR TITLE
change: clean leading quote from notification message

### PR DESCRIPTION
**Pull request by singularity-s0**

### CODE REVIEW

File: models/notification.go [hunk 1]
```diff
@@ -124,6 +126,18 @@ func (message *Notification) checkConfig() {
 	message.Recipients = newRecipient
 }
 
+func cleanNotificationDescription(content string) string {
+	newContent = strings.ReplaceAll(content, "\n", "")
+	newContent = regexp.MustCompile(`#{1,2}\d+`).ReplaceAllString(newContent, "")
+	newContent = regexp.MustCompile(`\${1,2}.*?\${1,2}`).ReplaceAllString(newContent, "[公式]")
+	newContent = regexp.MustCompile(`!\[\]\(dx_\S+\)`).ReplaceAllString(newContent, "[表情]")
+	newContent = regexp.MustCompile(`!\[.*?\]\(.*?\)`).ReplaceAllString(newContent, "[图片]")
```
Comment (w568w) [inline models/notification.go:134]: `MustCompile` is often used for global variable initialization. Have you considered made the compiled regex static for better performance?

**Review by w568w — CHANGES_REQUESTED** [review]

File: models/notification.go [hunk 2]
```diff
@@ -124,6 +126,18 @@ func (message *Notification) checkConfig() {
 	message.Recipients = newRecipient
 }
 
+func cleanNotificationDescription(content string) string {
+	newContent = strings.ReplaceAll(content, "\n", "")
```
Comment (ppolariss) [inline models/notification.go]: `newContent` is not declared before it’s used. You should declare the `newContent` variable first, like this:

```go
newContent := strings.ReplaceAll(content, "\n", "")
```

File: models/notification.go [hunk 3]
```diff
@@ -126,16 +126,23 @@ func (message *Notification) checkConfig() {
 	message.Recipients = newRecipient
 }
 
+var (
+    reHole    = regexp.MustCompile(`#{1,2}\d+`)
```
Comment (w568w) [inline models/notification.go:130]: A counterexample:

```
##12345 foo bar
```

will be transformed into:

```
# foo bar
```

It this expected?

**Comment by singularity-s0** [comment]
Why? Wouldn’t ‘#{1,2}’ match 2 # when possible?


On Nov 29, 2024, at 23:40, w568w ***@***.***> wrote:

﻿

@w568w commented on this pull request.

________________________________

In models/notification.go<https://github.com/OpenTreeHole/treehole_next/pull/168#discussion_r1863700493>:

> @@ -126,16 +126,23 @@ func (message *Notification) checkConfig() {
        message.Recipients = newRecipient
 }

+var (
+    reHole    = regexp.MustCompile(`#{1,2}\d+`)


A counterexample:

##12345 foo bar


will be transformed into:

# foo bar


It this expected?

**Comment by w568w** [comment]
> Why? Wouldn’t ‘#{1,2}’ match 2 # when possible?

I misread it. You can see that I have removed that comment.

File: models/notification.go (hunk 2 above)
Comment (singularity-s0) [reply models/notification.go]: This is fixed in the second commit.

File: models/notification.go [hunk 4]
```diff
@@ -124,6 +126,25 @@ func (message *Notification) checkConfig() {
...
+    reFormula = regexp.MustCompile(`\${1,2}.*?\${1,2}`)
```
Comment (w568w) [inline models/notification.go:131]: It does not match a multi-line string.

Play it here: <https://go.dev/play/p/O3g5Lz7W6Ta>

**Review by w568w — CHANGES_REQUESTED** [review]

**Review by w568w — APPROVED** [review]
LGTM.

File: models/notification.go [hunk 5]
```diff
@@ -124,6 +126,25 @@ func (message *Notification) checkConfig() {
...
 
+var (
+    reHole    = regexp.MustCompile(`#{1,2}\d+`)
+    reFormula = regexp.MustCompile(`(?s)\${1,2}.*?\${1,2}`)
+    reSticker = regexp.MustCompile(`!\[\]\(dx_\S+\)`)
+    reImage   = regexp.MustCompile(`!\[.*?\]\(.*?\)`)
+)
+
+func cleanNotificationDescription(content string) string {
+    newContent := strings.ReplaceAll(content, "\n", "")
```
Comment (JingYiJun) [inline models/notification.go:137]: 这个应该放最后吧

**Pull request merged**